examples/power_method.jl

```diff
@@ -0,0 +1,51 @@
+using Zygote
+using SparseArrays, LinearAlgebra
+using BenchmarkTools
+function power_max_eigen(A, x, target; niter=100)
+    for i=1:niter
+        x = A * x
+        x /= norm(x)
+    end
+    return abs(x' * target)
+end
+
+function power_min_eigen(A, x, target, λ; niter=100)
+    A = λ*I - A 
+    for i=1:niter
+        x = A * x
+        x /= norm(x)
+    end
+    return abs(x' * target)
+end
+
+
+A = sprand(100, 100, 0.1)
+x = randn(100)
+target = randn(100)
+λ = 2
+@btime max_ga_z, max_gx_z, max_gt_z = Zygote.gradient(power_max_eigen, A, x, target) # 3.694 ms (3198 allocations: 12.26 MiB)
+@btime min_ga_z, min_gx_z, min_gt_z = Zygote.gradient(power_min_eigen, A, x, target, λ) # 4.031 ms (3438 allocations: 12.62 MiB)
+
+using NiSparseArrays
+@btime max_ga, max_gx, max_gt = Zygote.gradient(power_max_eigen, A, x, target) # 1.413 ms (3398 allocations: 7.99 MiB)
+@btime min_ga, min_gx, min_gt = Zygote.gradient(power_min_eigen, A, x, target, λ) #  1.610 ms (3738 allocations: 8.62 MiB)
+
+using Test
+
+max_ga_z, max_gx_z, max_gt_z = Zygote.gradient(power_max_eigen, A, x, target)
+min_ga_z, min_gx_z, min_gt_z = Zygote.gradient(power_min_eigen, A, x, target, λ)
+max_ga, max_gx, max_gt = Zygote.gradient(power_max_eigen, A, x, target)
+min_ga, min_gx, min_gt = Zygote.gradient(power_min_eigen, A, x, target, λ)
+
+@testset begin
+    @testset "power method for max eigenvalue" begin
+        @test max_ga_z ≈ max_ga
+        @test max_gx_z ≈ max_gx
+        @test max_gt_z ≈ max_gt
+    end
+    @testset "power method for min eigenvalue" begin
+        @test min_ga_z ≈ min_ga
+        @test min_gx_z ≈ min_gx
+        @test min_gt_z ≈ min_gt
+    end
+end
```

Comment:
I'm not sure compare between `NiSparseArrays` and built-in `Zygote` is enough? Adding `FiniteDiff` or `ForwardDiff` for correctness is better? And should the example files be included in `runtest.jl`?

Reply:
Most demos don't need very formal tests; as long as it works and the results looks good, it looks good.

Reply:
Actually, I think using a benchmark to show off the performance would be good, this is the comparison with ForwardDiff (only sparse matrices are differentiated)

```julia
julia> @benchmark ForwardDiff.gradient(params->power_min_eigen(SparseMatrixCSC(A.m, A.n, A.colptr, A.rowval, params), x, target, λ), A.nzval)
BenchmarkTools.Trial: 83 samples with 1 evaluation.
 Range (min … max):  57.602 ms … 76.745 ms  ┊ GC (min … max): 5.56% … 7.17%
 Time  (median):     60.118 ms              ┊ GC (median):    8.24%
 Time  (mean ± σ):   60.719 ms ±  2.324 ms  ┊ GC (mean ± σ):  8.15% ± 0.65%

  █                                                            
  █▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁ ▁
  66.2 ms         Histogram: frequency by time        61.9 ms <

 Memory estimate: 179.57 MiB, allocs estimate: 18108.

julia> @benchmark Zygote.gradient(power_min_eigen, A, x, target, λ)
BenchmarkTools.Trial: 2485 samples with 1 evaluation.
 Range (min … max):  1.657 ms …   4.881 ms  ┊ GC (min … max):  0.00% … 39.49%
 Time  (median):     1.714 ms               ┊ GC (median):     0.00%
 Time  (mean ± σ):   2.007 ms ± 644.272 μs  ┊ GC (mean ± σ):  12.43% ± 16.99%

  █                                                            
  █▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁ ▁▁▁▁▁▁▁▁▁▁▁▁▁▁ ▁
  2.22 ms         Histogram: frequency by time        1.73 ms <

 Memory estimate: 8.79 MiB, allocs estimate: 4008.
```

Reply:
Just saw your benchmark, I think that is even better!

Reply:
Sorry for keep poping up new messages.
I would suggest using `A = sprand(5000, 5000, 0.001)` in benchmarks so that people will see the difference better, e.g. 
```julia
julia> @btime max_ga, max_gx, max_gt = Zygote.gradient($power_max_eigen, $A, $x, $target)
  5.500 s (4049 allocations: 18.75 GiB)

julia> using NiSparseArrays

julia> @btime max_ga, max_gx, max_gt = Zygote.gradient($power_max_eigen, $A, $x, $target)
  51.528 ms (5049 allocations: 220.84 MiB)
```